src/figuresClasses.ts

```diff
@@ -1,11 +1,81 @@
-export interface Figure {}
+type Colors = 'red' | 'green' | 'blue';
+type Shapes = `triangle` | `circle` | `rectangle`;
 
-export class Triangle implements Figure {}
+export interface Figure {
+  readonly shape: Shapes;
+  color: Colors;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  readonly shape: Shapes = `triangle`;
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalid triangle dimensions');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Invalid triangle dimensions');
+    }
+  }
+
+  getArea = (): number => {
```

Comment:
[CODE STYLE] - The `getArea` function doesn't need to accept any parameters. All necessary values are already available as object fields. Consider removing the parameter list and accessing the class fields directly.